src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -270,12 +273,15 @@ protected override void ExecuteCore()
                 {
                     string compositePDBImage = null;
                     string compositePDBRelativePath = null;
-                    if (IsTargetWindows && hasValidDiaSymReaderLib)
+                    if (IsTargetWindows)
                     {
-                        compositePDBImage = Path.ChangeExtension(compositeR2RImage, ".ni.pdb");
-                        compositePDBRelativePath = Path.ChangeExtension(compositeR2RImageRelativePath, ".ni.pdb");
+                        if (hasValidDiaSymReaderLib)
+                        {
+                            compositePDBImage = Path.ChangeExtension(compositeR2RImage, ".ni.pdb");
+                            compositePDBRelativePath = Path.ChangeExtension(compositeR2RImageRelativePath, ".ni.pdb");
+                        }
                     }
-                    else if (IsTargetLinux)
+                    else if ((ReadyToRunUseCrossgen2 && !_crossgen2IsVersion5) || IsTargetLinux)
```

Comment:
This code path is already under `if (Crossgen2Composite)`, so checking `ReadyToRunUseCrossgen2` is optional here.  You changed it in iteration 9 instead of the other place.

Reply:
OK, thanks for the heads-up. I believe that the composite build mode was only made official in .NET 6 so that we shouldn't actually need any extra check in the else block, I'll make the change in an extra commit.

Reply:
Sorry for the late reply.  On one hand, we default to the composite mode for .NET 5 if `PublishReadyToRunUseCrossgen2` is set: https://github.com/dotnet/sdk/blob/f5632207158c5800208d3482f5ed163e7dc54932/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.CrossGen.targets#L15-L16 On the other hand, generating the composite PDB in this method is guarded by the check at line 291, which requires targeting .NET 6 or higher: https://github.com/dotnet/sdk/blob/f5632207158c5800208d3482f5ed163e7dc54932/src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs#L284-L291 That means your change is correct even for the non-official composite mode in .NET 5.